### PR TITLE
Throttle email edits according to verification code TTL

### PR DESCRIPTION
**Pull request by iamacook**
This throttle email edit requests according to the TTL of verification codes. If an edit request occurs again within the TTL of the generated verification occurs, a `429` is returned.

- `EmailRepository['editEmail']` now includes a condition checking the presence of `verificationGeneratedOn` for the current email, then checking whether the current request is before or after the TTL and throwing accordingly.
- `EditTimespanError` error and associated `EmailEditTimespanExceptionFilter` exception filter have been included.
- Associated test coverage added.

### CODE REVIEW

File: src/domain/email/email.repository.ts [hunk 1]
```diff
@@ -170,6 +171,19 @@ export class EmailRepository implements IEmailRepository {
     const newEmail = new EmailAddress(args.emailAddress);
     const currentEmail = await this.emailDataSource.getEmail(args);
 
+    if (
+      currentEmail.verificationGeneratedOn &&
+      this._isEmailVerificationCodeValid(currentEmail)
```
Comment (hectorgomezv) [inline src/domain/email/email.repository.ts:176]: Not sure I get the purpose of this condition. The error seems to be thrown when the currentEmail has a previously generated verificationCode, and that verificationCode is valid, is that correct? Maybe I'm missing something.

Comment (iamacook) [reply src/domain/email/email.repository.ts:176]: > Not sure I get the purpose of this condition. The error seems to be thrown when the currentEmail has a previously generated verificationCode, and that verificationCode is valid, is that correct?

Yes, this prevents editing again within the TTL window as the `verificationCode` is only valid within it.

Comment (hectorgomezv) [reply src/domain/email/email.repository.ts:176]: Oh, you are right, I was misinterpreting the requirement 😅 Thanks!

**Review by hectorgomezv — APPROVED** [review]
Great! 🚀

**Review by hectorgomezv — APPROVED** [review]

**Pull request merged**